src/utils/signedUrlReplacer.ts

```diff
@@ -0,0 +1,43 @@
+import APIError from '@/app/api/core/exceptions/api'
+import { SupabaseActions } from '@/utils/SupabaseActions'
+
+export async function replaceImageSrc(htmlString: string, getSignedUrl: (filePath: string) => Promise<string>) {
+  const imgTagRegex = /<img\s+[^>]*src="([^"]+)"[^>]*>/g
```

Comment:
Could you add a few use cases / matches for this regex in comments above so we don't forget what this is and isn't supposed to match in the future